MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/BaseballSprayChartActivity.java

```diff
@@ -0,0 +1,298 @@
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.visibility;
+
+
+/**
+ * Use the Maps SDK's layer filters to explore baseball "spray chart" hit data. Filter by
+ * the result of the hit (single, home run, etc.) and/or by the type of
+ * hit (line drive, pop fly, etc.)
+ */
+public class BaseballSprayChartActivity extends AppCompatActivity {
+
+  private static final String HIT_LINE_LAYER_ID = "HIT_LINE_LAYER_ID";
+  private static final String BASEBALL_ICON_GEOJSON_SOURCE_ID = "BASEBALL_ICON_GEOJSON_SOURCE_ID";
+  private static final String BASEBALL_SYMBOL_LAYER_ID = "BASEBALL_SYMBOL_LAYER_ID";
+  private static final String BASEBALL_ICON_ID = "BASEBALL_ICON_ID";
+  private static final String VECTOR_SOURCE_ID = "VECTOR_SOURCE_ID";
+  private static final LatLng BOUND_CORNER_NW = new LatLng(37.77942401073674, -122.3878240585327);
+  private static final LatLng BOUND_CORNER_SE = new LatLng(37.77709202770888, -122.3908495903015);
+  private static final LatLngBounds RESTRICTED_BOUNDS_AREA = new LatLngBounds.Builder()
+      .include(BOUND_CORNER_NW)
+      .include(BOUND_CORNER_SE)
+      .build();
+  private MapView mapView;
+  private MapboxMap mapboxMap;
+  private VectorSource vectorSource;
+  private LineLayer hitLineLayer;
+  private SymbolLayer baseballSymbolLayer;
+  private ArrayList<String> hitResultSpinnerOptionList;
+  private boolean hitTypeHasBeenSelected = false;
+  private int selectedHitResult = 0;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_labs_baseball_spray_chart);
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(@NonNull MapboxMap mapboxMap) {
+        mapboxMap.setStyle(Style.SATELLITE,
+            new Style.OnStyleLoaded() {
+              @Override
+              public void onStyleLoaded(@NonNull Style style) {
+                BaseballSprayChartActivity.this.mapboxMap = mapboxMap;
+                mapboxMap.setLatLngBoundsForCameraTarget(RESTRICTED_BOUNDS_AREA);
+                initHitLineLayer(style);
+                initBaseTypeSpinner();
+                initBaseballSymbolLayer(style);
+                initBaseballIconCheck();
+              }
+            });
+      }
+    });
+  }
+
+  private void initHitLineLayer(Style loadedStyle) {
+    vectorSource = new VectorSource(VECTOR_SOURCE_ID,
+        "mapbox://appsatmapboxcom.ck6ybxaey1hue2lnozejkugfz-1n5cb");
```

Comment:
maybe explain in a comment or class coumentation what this vector source is? How did you set this up so users can do this themselves?

Reply:
Done

```
    This example loads data from a Mapbox Tileset.
    A tileset is a collection of raster or vector data broken
    up into a uniform grid of square tiles. A tileset can have
    data within it.

    If you use a VectorSource, make sure you use the `setSourceLayer()`
    method, as is done below. This tells the Maps SDK to drill one level
    deeper into the tileset and reference the correct data for the layer.
    If you've created your own Mapbox account, you can find tilesets at
    https://studio.mapbox.com/tilesets. You can make a GeoJSON-based
    dataset at https://studio.mapbox.com/datasets, have Mapbox Studio
    export the dataset to a tileset, and then use the tileset in your
    project.

    https://docs.mapbox.com/mapbox-gl-js/style-spec/sources/#vector
    has more about a VectorSource. VectorSource objects are also used
    in other examples in this app.
```